docs/orchestrating-elastic-stack-applications/logstash.asciidoc

```diff
@@ -874,12 +874,147 @@ spec:
+----
+spec:
+  podTemplate:
+    spec:
+      initContainers:
+      - name: install-plugins
+        command: ["/bin/sh"]
+        args: ["-c", "bin/logstash-plugin install logstash-filter-tld"]
+----
+
+The init container inherits:
+
+* The image of the main container image, if one is not explicitly set.
+* The volume mounts from the main container unless a volume mount with the same name and mount path is present in the init container definition.
+* The Pod name and IP address environment variables.
+
+
+=== Adding a plugin using a custom image
+
+This example installs the same plugin, but uses a custom image to install the plugin.
+
+First, create a Dockerfile based on the {ls} Docker image, running `bin/logstash-plugin install` to install the appropriate plugin(s):
+
+[subs="attributes,+macros,callouts"]
+----
+FROM docker.elastic.co/logstash/logstash:{version}
+
+RUN bin/logstash-plugin install logstash-filter-tld
+----
+
+Then after building and deploying the custom image (refer to <<{p}-custom-images>> for more details), include it in the {ls} manifest:
+
+[source,yaml]
+----
+spec:
+  count: 1
+  version: {version} <1>
+  image: <CUSTOM_IMAGE>
+----
+<1> Providing the correct version is always required as ECK reasons about APIs and capabilities available to it based on the version field.
+
+
+[id="({p}-logstash-jar-files"]
+== Working with plugins that require additional files
+
+Running {ls} may require additional files, such as JAR files needed to load JDBC drivers when using a JDBC or JMS plugin. To add these files, there are two options available - using an `initContainer` to add files before the main container start, or creating a custom Docker image that includes the required files. Refer to <<{p}-bundles-plugins>> for a run down of which option might be most suitable for you.
+
+
+=== Adding files using an initContainer
+
+This example creates an `initContainer` to download a PostgreSQL JDBC driver JAR file, and place it in a volume mount accessible to the main `container`, and then use it in a JDBC input in the pipeline configuration.
+
+[source,yaml]
+----
+spec:
+  podTemplate:
+    spec:
+      initContainers:
+      - name: download-postgres
+        command: ["/bin/sh"]
+        args: ["-c", "curl -o /data/postgresql.jar -L https://jdbc.postgresql.org/download/postgresql-42.6.0.jar"]
+        volumeMounts:
+          - name: workdir
+            mountPath: /data
+      containers:
+        - name: logstash
+          volumeMounts:
+            - name: workdir
+              mountPath: /usr/share/logstash/jars <1>
+  pipelines:
+    - pipeline.id: main
+      config.string: |
+        input {
+          jdbc {
+             jdbc_driver_library => "/usr/share/logstash/jars/postgresql.jar"
+             jdbc_driver_class => "org.postgresql.Driver"
+             <2>
+          }
+        }
+----
+<1> Referring to the external file should match the `mountPath` of the `container`
+<2> Rest of plugin configuration goes here
```

Comment:
```suggestion
<2> Remainder of plugin configuration goes here
```

Reply:
Maybe "remainder" instead of "rest" because "rest" can have multiple meanings.  
Or "Remaining plugin configuration goes here." 
You make the call.